sources/api/netdog/src/networkd/config/netdev.rs

```diff
@@ -144,4 +147,175 @@ impl NetDevConfig {
 
         Ok(path)
     }
+
+    // The following *mut() methods are private and primarily meant for use by the NetDevBuilder.
+    // They are convenience methods to access the referenced structs (which are `Option`s) since
+    // they may need to be accessed in multiple places during the builder's construction process.
+    // (And no one wants to call `get_or_insert_with()` everywhere)
+    fn vlan_mut(&mut self) -> &mut VlanSection {
+        self.vlan.get_or_insert_with(VlanSection::default)
+    }
+
+    fn bond_mut(&mut self) -> &mut BondSection {
+        self.bond.get_or_insert_with(BondSection::default)
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+//
+/// The builder for `NetDevConfig`.
+//
+// Why a builder?  Great question.  As you can see below, some logic is involved to translate
+// config struct fields to a valid NetDevConfig.  Since `NetDevConfig` will be created by multiple
+// devices (bonds and VLANs to start), it makes sense to centralize that logic to avoid
+// duplication/mistakes.  Using a builder means type parameters can be used to limit available
+// methods based on the device being created.  Putting the type parameter on the builder and not
+// NetDevConfig avoids proliferating the type parameter everywhere NetDevConfig may be used.
+#[derive(Debug)]
+pub(crate) struct NetDevBuilder<T: Device> {
+    netdev: NetDevConfig,
+    spooky: PhantomData<T>,
+}
+
+impl<T: Device> NetDevBuilder<T> {
+    // The builder stores and directly manipulates the NetDevConfig being created to avoid storing
+    // yet another copy of the data supplied to the builder.  Generally builders have a `build()`
+    // method that returns the type being built; stick with that pattern here.  This also avoids
+    // proliferating the type parameters and provides the opportunity to add any additional
+    // creation logic needed in the future.
```

Comment:
Sorry I should have been a bit more clear.  I was alluding to the fact that builders sometimes store all of the data for the type it is constructing, and then do all the processing during a `build()` method.  Instead here we are directly manipulating the underlying type rather than store a copy of the data.

Reply:
I think I'll remove it - it does seem a bit superfluous.

Reply:
Yeah, it seems overly geared toward the PR, i.e. to convince reviewers that the design is right, which becomes somewhat moot after it is merged.